web-api/src/utils.rs

```diff
@@ -33,6 +31,31 @@ impl From<&str> for RelativePathBuf {
     }
 }
 
+impl Serialize for RelativePathBuf {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        if let Some(path) = self.relative().as_os_str().to_str() {
+            path.serialize(serializer)
+        } else {
+            Err(serde::ser::Error::custom(
+                "Can't serialize non string path.",
+            ))
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for RelativePathBuf {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = String::deserialize(deserializer)?;
```

Comment:
We can't use `<&str>::deserialize` due to how `serde(flatten)` works, the bug btw. was also caused by the subtle interactions of default values and `serde(flatten)`.